File: sorting/selectionSort/selectionSort.py
arr1 = [9,1,4,6,2,3,6,76,23,1]
arr2 = [91,123,23,523,12,5]
arr3 = [9,8,7,6,5,4,3,2,1]

def selectionSort(arr):

    # iterate over indices in arr
    for i in range(len(arr)):

        # variable to store index of smallest value in arr
        smallest_index = i

        # iterate over all indices after current index(i)
        for j in range(i+1, len(arr)):

            # if the value at index j is less than the value at the index stored in the smallest_index variable
            if arr[j] < arr[smallest_index]:

                # updated smallest_index to store the index j
                smallest_index = j

        # swap the values at the current index (i) & the value at the smallest_indexx
        arr[i], arr[smallest_index] = arr[smallest_index], arr[i]

    # return arr
    return arr

print(selectionSort(arr1))
print(selectionSort(arr2))
print(selectionSort(arr3))

File: sorting/selectionSort/selectionSort.js
arr1 = [9,1,4,6,2,3,6,76,23,1]
arr2 = [91,123,23,523,12,5]
arr3 = [9,8,7,6,5,4,3,2,1]

function selectionSort(arr) {
    // iterate over the array
    for (i=0; i < arr.length; i++) {

        // have variable for smallest index, default at current index in our foor loop
        let smallestIndex = i

        // iterate over the rest of the arr, from the next index to the end
        for (j=i+1; j < arr.length; j++) {

            // if the value at index j is smaller than the value at index of our current smallest index
            if (arr[j] < arr[smallestIndex]) {

                // update the index stored in smallestIndex to the index at j
                smallestIndex = j
            }
        }

        // swap the values of the value at the index of our smallest value and the index in our current iteration of i. This moves the smallest values to the front of the list
        const temp = arr[smallestIndex]
        arr[smallestIndex] = arr[i]
        arr[i] = temp
    }

    // return the sorted arr
    return arr
}

console.log(selectionSort(arr1))
console.log(selectionSort(arr2))
console.log(selectionSort(arr3))